clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -1224,8 +1233,8 @@ raw2trace_t::process_syscall_pt(raw2trace_thread_data_t *tdata, uint64_t syscall
                                 .addr = sysnum };
     entries.push_back(end_entry);
     if (entries.size() == 2) {
-        tdata->error = "No trace entries generated from PT data";
-        return false;
+        accumulate_to_statistic(tdata, RAW2TRACE_STAT_SYSCALL_TRACES_DECODE_EMPTY, 1);
+        return true;
```

Comment:
Not sure I understand: you're now considering fatal or serious errors to not be errors?  But I don't see the caller's code changing?  So PT conversion is always a success now and you have to look at these stats to see whether it produced anything usable?

Reply:
> you're now considering fatal or serious errors to not be errors?

If you're referring to this particular one for empty traces: I'm not sure yet it's really a "fatal or serious" error or if some syscalls may be immediately interrupted or something thus producing no trace.

> you have to look at these stats to see whether it produced anything usable?

Yes.

Reply:
Is this the aforementioned missing syscall trace?

Reply:
Maybe there are two different things being discussed here.

If a converted syscall trace is empty because we couldn't decode the PT raw trace at all: note that this would already have returned above at L1199.

if a converted syscall trace is empty because there wasn't anything to decode because there was no PT data: this is the case handled by this if-block.

```
 I'm not sure yet it's really a "fatal or serious" error or if some syscalls may be immediately interrupted or something thus producing no trace.
```